kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaTopicManager.java

```diff
@@ -226,26 +219,14 @@ private void lookupBroker(String topicName,
                     retFuture.complete(pair.getLeft());
                 })
                 .exceptionally(th -> {
-                    long waitTimeMs = backoff.next();
-
-                    if (backoff.isMandatoryStopMade()) {
-                        log.warn("[{}] getBroker for topic {} failed, retried too many times {}, return null."
-                                        + " throwable: ", requestHandler.ctx.channel(), topicName, waitTimeMs, th);
-                        retFuture.complete(null);
-                    } else {
-                        log.warn("[{}] getBroker for topic failed, will retry in {} ms. throwable: ",
-                            topicName, waitTimeMs, th);
-                        requestHandler.getPulsarService().getExecutor()
-                            .schedule(() -> lookupBroker(topicName, backoff, retFuture),
-                                waitTimeMs,
-                                TimeUnit.MILLISECONDS);
-                    }
+                    log.warn("[{}] getBroker for topic failed. throwable: ",
+                            topicName, th);
+                    retFuture.complete(null);
                     return null;
                 });
         } catch (PulsarServerException e) {
             log.error("[{}] getTopicBroker for topic {} failed get pulsar client, return null. throwable: ",
                 requestHandler.ctx.channel(), topicName, e);
-            retFuture.complete(null);
```

Comment:
So if the code reached here, the `retFuture` would be a never completed `CompletableFuture`? I'm afraid it's not a good behavior.

Reply:
fixed